### PR TITLE
[MM-54818] Convert system_permission_gate.tsx from Class Component to Function Component

### PR DESCRIPTION
**Pull request by escofresco**
#### Summary
Refactors SystemPermissionGate class to function. Tested by running `npm test -u -t /webapp/channels/src/components/permissions_gates/system_permission_gate/system_permission_gate.test.tsx`.

Referenced [blog post on refactoring Default Props](https://blog.bitsrc.io/understanding-react-default-props-5c50401ed37d)
#### Ticket Link
Fixes https://github.com/mattermost/mattermost/issues/24779
Jira https://mattermost.atlassian.net/browse/MM-54818

#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: webapp/channels/src/components/permissions_gates/system_permission_gate/system_permission_gate.tsx [hunk 1]
```diff
@@ -25,18 +25,15 @@ type Props = {
...
-
-    render() {
-        if (this.props.hasPermission && !this.props.invert) {
-            return this.props.children;
-        }
-        if (!this.props.hasPermission && this.props.invert) {
-            return this.props.children;
-        }
-        return null;
+export default function SystemPermissionGate(props: Props) {
```
Comment (larkox) [inline webapp/channels/src/components/permissions_gates/system_permission_gate/system_permission_gate.tsx:28]: Just for consistency, we prefer to have arrow functions and the export separated at the end. i.e.:
```
const SystemPermissionGate = (props: Props) => {
   ...
}
export default SystemPermissionGate;

File: webapp/channels/src/components/permissions_gates/system_permission_gate/system_permission_gate.tsx [hunk 2]
```diff
@@ -25,18 +25,15 @@ type Props = {
...
+    if (props.hasPermission && !props.invert) {
+        return props.children;
+    }
+    if (!props.hasPermission && props.invert) {
+        return props.children;
     }
+    return null;
 }
+SystemPermissionGate.defaultProps = {
```
Comment (larkox) [inline webapp/channels/src/components/permissions_gates/system_permission_gate/system_permission_gate.tsx:37]: For function components, you wouldn't use defaultProps. The most common approach we use is to expand the props on the function and assign there the default. In this case...

```
const SystemPermissionGate = ({
   permissions,
   hasPermissions,
   invert = false,
   children, 
}: Props) => {
```

**Review by larkox — CHANGES_REQUESTED** [review]
Thank you for the contribution! Just a few minor style changes.

**Review by sinansonmez — APPROVED** [review]
LGTM 🚀 thanks

**Review by larkox — APPROVED** [review]
LGTM. Thanks!

**Comment by larkox** [comment]
@escofresco There seem to be some issues with lint and types. You can see the details in the checks. Let me know if you have any problem fixing those.

**Comment by larkox** [comment]
@escofresco There seem to be still some errors around lint, types and tests. Can you take a look at them?

**Comment by escofresco** [comment]
Will investigate right now

**Comment by escofresco** [comment]
@larkox Style issues should be fixed now, runs fine locally, also combined
```typescript   
if (hasPermission && !invert) {
    …
}
if (!hasPermission && invert) {
    …
}
```
into 
```typescript
if (hasPermission !== invert) {
    …
}
``` 
because seems like XOR to me. Let me know if you'd like that to be reverted though

**Comment by escofresco** [comment]
Tests were also failing locally until `--update-snapshot` because the original test snapshots contained `invert={false}` which were unmatched when the invert default prop was set as default destructure argument in `SystemPermissionGate`. Otherwise, I wasn't able to find a way for the snapshots to match without using `SystemPermissionGate.defaultProps = { invert: false }`

**Comment by lindy65** [comment]
Hi @escofresco - please run `npm run test -- -u` for failing tests :)

**Comment by escofresco** [comment]
@lindy65 added the new snapshot

**Review by lindy65 — APPROVED** [review]
Thanks @escofresco - looks good to me 👍 

cc// @larkox - good to go after E2E run

**Comment by larkox** [comment]
@lindy65 Can you double check the cases where the e2e are failing? It should not be related, but better safe than sorry.

**Comment by lindy65** [comment]
Checked the failing E2E tests @larkox - they are not related to this PR 👍

**Pull request merged**